modules/op-running-unprivileged-builds-in-a-container.adoc

```diff
@@ -0,0 +1,131 @@
+// Module included in the following assemblies:
+//
+// * cicd/pipelines/running-workloads-and-buildah-as-user-namespaces-on-openshift-pipelines.adoc
+:_content-type: PROCEDURE
+
+[id="op-running-unprivileged-builds-in-a-container_{context}"]
+= Running unprivileged builds in a container
+
+You can run builds as a non-root user inside a container to restrict specific container activities and enhance security.
+
+.Prerequisites
+
+. Ensure that the container is configured properly by associating the `/etc/subuid` and `/etc/subgid` files with the latest `shadow-utils` package.
+
+. Create a `ConfigMap` link:https://kubernetes.io/docs/concepts/storage/volumes/#configmap[storage volume] that the build can use as a workspace.
+
+.Procedure
+
+. Use the publicly available link:https://catalog.redhat.com/software/containers/ubi8/buildah/602686f7b16b1eb2e30807ee?container-tabs=dockerfile[Containerfile] image, which is based on the Red Hat Universal Base Image (UBI).
+
+. Force run the container as the “build” user to prevent it from being run randomly without the setup necessary to run as a non-root user.
```

Comment:
the build user is only specific to the buildah container, if you setup your own user, you need to force run to this user which has the subuid/subgid setup.